e2e/test/iothub/service/RegistryManagerExportDevicesTests.cs

```diff
@@ -163,55 +155,57 @@ public async Task RegistryManager_ExportDevices(StorageAuthenticationType storag
             RegistryManager registryManager,
             Uri containerUri)
         {
-            int tryCount = 0;
-
             ManagedIdentity identity = isUserAssignedMsi
                 ? new ManagedIdentity
                 {
                     UserAssignedIdentity = TestConfiguration.IoTHub.UserAssignedMsiResourceId
                 }
                 : null;
 
-            JobProperties exportJobResponse = JobProperties.CreateForExportJob(
+            JobProperties jobProperties = JobProperties.CreateForExportJob(
                 containerUri.ToString(),
                 true,
                 devicesFileName,
                 storageAuthenticationType,
                 identity);
-            exportJobResponse.IncludeConfigurations = true;
-            exportJobResponse.ConfigurationsBlobName = configsFileName;
+            jobProperties.IncludeConfigurations = true;
+            jobProperties.ConfigurationsBlobName = configsFileName;
 
-            while (tryCount < MaxIterationWait)
+            var sw = Stopwatch.StartNew();
+
+            while (!jobProperties.IsFinished)
             {
                 try
                 {
-                    exportJobResponse = await registryManager.ExportDevicesAsync(exportJobResponse).ConfigureAwait(false);
+                    jobProperties = await registryManager.ExportDevicesAsync(jobProperties).ConfigureAwait(false);
+                    if (!string.IsNullOrWhiteSpace(jobProperties.FailureReason))
+                    {
+                        Logger.Trace($"Job failed due to {jobProperties.FailureReason}");
+                    }
                     break;
                 }
                 // Concurrent jobs can be rejected, so implement a retry mechanism to handle conflicts with other tests
-                catch (JobQuotaExceededException) when (++tryCount < MaxIterationWait)
```

Comment:
Rely on the test timeout, not an arbitrary number of retries (which when multiplied by the sleep gives a not-obvious test timeout).